plugins/newsdownloader.koplugin/main.lua

```diff
@@ -135,32 +136,75 @@ function NewsDownloader:loadConfigAndProcessFeeds()
             UIManager:show(info)
             -- processFeedSource is a blocking call, so manually force a UI refresh beforehand
             UIManager:forceRePaint()
-            self:processFeedSource(url, tonumber(limit))
+            self:processFeedSource(url, tonumber(limit), unsupported_feeds_urls)
             UIManager:close(info)
         else
             logger.warn('NewsDownloader: invalid feed config entry', feed)
         end
     end
 
-    UIManager:show(InfoMessage:new{
-        text = _("Downloading news finished."),
-        timeout = 1,
-    })
+    if #unsupported_feeds_urls <= 0 then
+        UIManager:show(InfoMessage:new{
+            text = _("Downloading news finished."),
+            timeout = 1,
+        })
+    else
+        local unsupported_urls = ""
+        for k,url in pairs(unsupported_feeds_urls) do
+            unsupported_urls = unsupported_urls .. url
+            if k ~= #unsupported_feeds_urls then
+                unsupported_urls = unsupported_urls .. ", "
+            end
+        end
+        UIManager:show(InfoMessage:new{
+            text = T(_("Downloading news finished. Could not process some feeds. Unsupported format in: %1"), unsupported_urls)
+        })
+    end
 end
 
-function NewsDownloader:processFeedSource(url, limit)
+function NewsDownloader:processFeedSource(url, limit, unsupported_feeds_urls)
     local resp_lines = {}
     http.request({ url = url, sink = ltn12.sink.table(resp_lines), })
     local feeds = deserializeXMLString(table.concat(resp_lines))
-    if not feeds then return end
-    if not feeds.rss or not feeds.rss.channel
-            or not feeds.rss.channel.title or not feeds.rss.channel.item then
-        logger.info('NewsDownloader: Got invalid feeds', feeds)
+    if not feeds then
+        table.insert(unsupported_feeds_urls, url)
         return
     end
 
+    local is_rss = feeds.rss and feeds.rss.channel and feeds.rss.channel.title and feeds.rss.channel.item;
+    local is_atom = feeds.feed and feeds.feed.title and feeds.feed.entry.title and feeds.feed.entry.link;
+
+    if not is_rss and not is_atom then
+        table.insert(unsupported_feeds_urls, url)
+        return
+    end
+
+    if is_atom then
+        self:processAtom(feeds, limit);
+    else
+        self:processRSS(feeds, limit);
+    end
```

Comment:
This whole block can be optimized to the following to remove redundant if comparisons:

```lua
if feeds.rss and feeds.rss.channel and feeds.rss.channel.title and feeds.rss.channel.item then
    self:processRSS(feeds, limit)
elseif feeds.feed and feeds.feed.title and feeds.feed.entry.title and feeds.feed.entry.link then
    self:processAtom(feeds, limit)
else
    table.insert(unsupported_feeds_urls, url)
    return
end
```

Also notice `;` is not needed in LUA.